src/components/Images.vue

```diff
@@ -0,0 +1,283 @@
+
+const { ipcRenderer } = require('electron');
+const K8s = require('../k8s-engine/k8s');
+
+export default {
+  components: {
+    ButtonDropdown, Checkbox, SortableTable
+  },
+  props:      {
+    images: {
+      type:     Array,
+      required: true,
+    },
+    k8sState: {
+      type:    Number,
+      default: K8s.State.STOPPED,
+    },
+    showAll: {
+      type:    Boolean,
+      default: false,
+    },
+  },
+
+  data() {
+    return {
+      kimRunningCommand: null,
+      headers:           [
+        {
+          name:  'imageName',
+          label: 'IMAGE',
+          sort:  ['imageName', 'tag', 'imageID'],
+        },
+        {
+          name:  'tag',
+          label: 'TAG',
+          sort:  ['tag', 'imageName', 'imageID'],
+        },
+        {
+          name:  'imageID',
+          label: 'IMAGE ID',
+          sort:  ['imageID', 'imageName', 'tag'],
+        },
+        {
+          name:  'size',
+          label: 'SIZE',
+          sort:  ['size', 'imageName', 'tag'],
+        },
+      ],
+      imageToBuild:                     '',
+      imageToPull:                      '',
+      imageManagerOutput:               '',
+      keepImageManagerOutputWindowOpen: false,
+      fieldToClear:                     null,
+    };
+  },
+  computed: {
+    rows() {
+      if (this.showAll) {
+        return this.images;
+      }
+
+      return this.images.filter(this.isDeletable);
+    },
+    k8sIsRunning() {
+      return this.k8sState === K8s.State.STARTED;
+    },
+    showImageManagerOutput() {
+      return !!this.kimRunningCommand || this.keepImageManagerOutputWindowOpen;
+    },
+    imageManagerProcessIsFinished() {
+      return !this.kimRunningCommand;
+    },
+    imageToBuildButtonDisabled() {
+      return this.showImageManagerOutput || this.imageToBuild.length === 0 || !this.imageToBuild.includes(':');
+    },
+    imageToPullButtonDisabled() {
+      return this.showImageManagerOutput || this.imageToPull.length === 0;
+    },
+  },
+
+  mounted() {
+    ipcRenderer.on('kim-process-ended', (event, status) => {
+      this.handleProcessEnd(status);
+    });
+    ipcRenderer.on('kim-process-output', (event, data, isStderr) => {
+      this.appendImageManagerOutput(data, isStderr);
+    });
+  },
+
+  methods: {
+    buttonOptions(row) {
+      const items = [];
+
+      if (this.isPushable(row)) {
+        items.push({
+          label:  'Push',
+          action: this.doPush,
+          value:  row,
+        });
+      }
+      if (this.isDeletable(row)) {
+        items.push({
+          label:  `Delete`,
+          action: this.deleteImage,
+          value:  row,
+        });
+      }
+
+      return items;
+    },
+    appendImageManagerOutput(data, isStderr) {
+      const outputWindow = this.$refs.outputWindow;
+
+      this.imageManagerOutput += data;
+      if (outputWindow) {
+        outputWindow.scrollTop = outputWindow.scrollHeight;
+      }
+    },
+    closeTheOutputWindow(event) {
```

Comment:
```suggestion
    closeOutputWindow(event) {
```
The article is redundant.

Reply:
ok